### ISSUE TITLE
[2018 ECCV] Modeling Visual Context is Key to Augmenting Object Detection Datasets

### ISSUE THREAD

**Issue by Jasonlee1995** [Jasonlee1995/AI_Papers#4]
해당 논문의 핵심만 이야기하면

Object detection 분야에서 사용될 수 있는 새로운 data augmentation을 제시함

해당 방법은 segmentation label이 있는 object를 context를 고려하여 이미지에 붙여넣기

![image](https://user-images.githubusercontent.com/49643709/110403616-c01af380-80c0-11eb-9208-2b99ac9f2aa0.png)

Instance detection 분야에서 제안된 data augmentation인 'Cut, Paste and Learn'과 상당히 유사하나 'Cut, Paste and Learn'은 random하게 붙여넣고 해당 논문은 학습을 통해 삽입 장소를 추려 붙여넣음

새롭게 제안한 이유는 object detection 분야에서 random placement를 하면 성능이 오히려 저하되기 때문

Data augmentation과 관련한 논문이기에, object detection 모델을 고정해두고 비교하는 형식으로 논문이 진행됨

전체적인 pipeline은 다음과 같음

![image](https://user-images.githubusercontent.com/49643709/110403964-5c44fa80-80c1-11eb-9234-8db697b71462.png)

**Comment by Jasonlee1995** [comment]
'Instance와의 context를 고려하여 붙여넣을 장소를 image에서 어떻게 찾나'에 대하여 다음과 같은 흐름으로 진행된다

1. Contextual data generation
2. Model training
3. Selection of object locations at test time
4. Blending objects in their environment

## 1. Contextual data generation
- Dataset : images with bounding box and category annotations (paper used PASCAL VOC 2012)
- Process of Generating Contextual Data
  * Joint distribution : positive samples로 aspect ratio a와 relative scale s를 변수로 하는 30 * 30 bins normalized histogram 생성
  * Positive samples : ground truth bbox를 포함하도록 random crop (ground truth bbox는 masking)
  * Negative samples : class bbox와 iou가 0.3 안넘도록 joint distribution을 이용하여 random crop (class bbox는 masking)
  * Negative samples를 3배 더 sample하여 contextual images 구축

## 2. Model training
- 모델 : ResNet 50 (pre-trained weights on ImageNet)
- Input : 1번 과정을 통해 생성된 contextual images (resize to 300 * 300)
- Output : label (num of classes + 1 for background)

## 3. Selection of object locations at test time
- Joint distribution을 이용하여 200개의 bbox random 생성
- 200개의 bbox에 대하여 각각의 bbox를 포함하는 더 큰 사이즈의 candidate bbox random 생성
- Candidate bbox를 masking 처리
- Masked candidate bbox 중 model output score이 0.8 이상인 경우만 선별

## 4. Blending objects in their environment
- 3번에서 선별된 bbox에 model output score이 높았던 class instance 붙여넣기
- 붙여넣을 때 blending techniques 중 1개 random 적용 (Gaussian blur, linear blur, motion blur, original)

**Comment by Jasonlee1995** [comment]
결과는 다음과 같음

![image](https://user-images.githubusercontent.com/49643709/110450562-ab614e80-8106-11eb-8494-91079612949f.png)
![image](https://user-images.githubusercontent.com/49643709/110450578-aef4d580-8106-11eb-84ad-46de2951bb02.png)
![image](https://user-images.githubusercontent.com/49643709/110450609-b61be380-8106-11eb-9b0d-88f4b2617e7d.png)
